### PR TITLE
Adds the new GPG key to the preinst script

### PR DESCRIPTION
**Pull request by gmmeyer**
Adds the new GPG key to the preinst script for Zypp and Yum based systems.

### CODE REVIEW

File: package-scripts/datadog-agent/preinst [hunk 1]
```diff
@@ -46,10 +46,27 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
       useradd -r -M -g dd-agent -d $INSTALL_DIR -s /bin/sh \
         -c "Datadog Agent" dd-agent
     # Stop the old agent before installing
-    if [ -f "/etc/init.d/datadog-agent" ]; then
```
Comment (degemer) [inline package-scripts/datadog-agent/preinst]: Why change that ?

File: package-scripts/datadog-agent/preinst [hunk 2]
```diff
@@ -46,10 +46,27 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
...
+    if [ -f "/etc/" ]; then
       /etc/init.d/datadog-agent stop || true
     fi
 
+    # Updates the yum repo to add the new GPG key for rhel based installations
```
Comment (degemer) [inline package-scripts/datadog-agent/preinst]: We could maybe add the same disclaimer as Debian: https://github.com/DataDog/dd-agent-omnibus/blob/40a28d59096d1ba606494ad580ced856f6e2ad35/package-scripts/datadog-agent/preinst#L35-L38

File: package-scripts/datadog-agent/preinst (hunk 1 above)
Comment (gmmeyer) [inline package-scripts/datadog-agent/preinst]: I think I meant to copy something and cut instead, but not really sure how I did it, either way it's back.

**Comment by gmmeyer** [comment]
@degemer I agree with you that this is problematic. A sha5 might work, it's probably the preferred solution here. What should we do for people who have a custom repo?

**Comment by degemer** [comment]
Could we just append the new key at the end of the file, or is it not possible with the file format?
Side note: I thought `> file` failed when the file already existed.

**Comment by gmmeyer** [comment]
woah thanks for following up! 

@degemer it does not, it overwrites

**Comment by degemer** [comment]
Aaaand it seems that I always had noclobber option turned on.
So for now, I think it's a breaking change for people using their own repos (since it's going to overwrite their urls). But it's probably find since they're likely using Chef/Puppet/... to manage this file.
Summary:
- 95% of people with the default -> overwriting is fine
- 5% remaining -> maybe `grep datadoghq.com` ? If it's there, overwrite as usual, if not, they're using a custom repo and we shouldn't touch the file (hopefully managed by Chef/Puppet so already having our key).

**Comment by gmmeyer** [comment]
this is dead I guess! closing

**Pull request closed without merging**